### PR TITLE
fix(docs) Add missing starter info to Gatsby/starters #22503

### PR DESCRIPTION
**Pull request by Edmund-Adewu**
## Description

This PR is meant to close issue #22503. The starter subsection contained only 3 official starters when there were in fact 5 listed on the official Gatsby website.
### Documentation

The relevant documentation exists in the gatsby/starters folder, specifically the README.md, blog theme and workspace theme subsections.
## Related Issues
The PR addresses the missing demos in #22503.

### CODE REVIEW

File: starters/gatsby-starter-blog-theme/.prettierrc [hunk 1]
```diff
@@ -1,7 +1,4 @@
 {
-  "endOfLine": "lf",
```
Comment (gillkyle) [inline starters/gatsby-starter-blog-theme/.prettierrc:2]: I don't think the changes here (on all files under `starters/gatsby-starter-blog-theme/*`) are necessary.

Could you please revert the changes in these files so that the starter remains as it was? It keeps the history of the files cleaner if we reduce the churn of unnecessary changes. 🙂

File: docs/starters.yml [hunk 2]
```diff
@@ -90,6 +90,14 @@
     - Includes plugins for easy, beautiful typography
     - Includes React Helmet to allow editing site meta tags
     - Includes plugins for offline support out of the box
+- url: https://github.com/gatsbyjs/gatsby/tree/master/starters/gatsby-starter-blog-theme
```
Comment (gillkyle) [inline docs/starters.yml:93]: There's a mirrored repository for this starter: https://github.com/gatsbyjs/gatsby-starter-blog-theme 🙂 

```suggestion
- url: https://github.com/gatsbyjs/gatsby-starter-blog-theme
```

File: docs/starters.yml [hunk 3]
```diff
@@ -90,6 +90,14 @@
...
+  repo: https://github.com/gatsbyjs/gatsby/tree/master/starters/gatsby-starter-blog-theme
```
Comment (gillkyle) [inline docs/starters.yml:94]:
```suggestion
  repo: https://github.com/gatsbyjs/gatsby-starter-blog-theme
```

Same thing here

File: docs/starters.yml [hunk 4]
```diff
@@ -90,6 +90,14 @@
...
+  description: official blog theme
+  tags:
+    - Official
+    - Blog
+  features:
+    - Includes plugin for adjustment of background lighting
```
Comment (gillkyle) [inline docs/starters.yml:100]: I think there are more features we could add here. The blog theme has:
- blog posts set up from a content folder
- styling done with Theme-UI showing how to shadow a theme file

...and more stuff set up for you. It would be good to highlight what we can, if this is really to help showcase these official starters we should make them shine!

File: docs/starters.yml [hunk 5]
```diff
@@ -98,6 +106,13 @@
   features:
     - Comes with React Helmet for adding site meta tags
     - Includes plugins for offline support out of the box
+- url: https://github.com/gatsbyjs/gatsby-starter-theme-workspace
+  repo: https://github.com/gatsbyjs/gatsby-starter-theme-workspace
+  description: official workspace theme
+  tags:
+    - Official
+  features:
+    - Minimalist plugin to make content stand out
```
Comment (gillkyle) [inline docs/starters.yml:115]: This isn't entirely accurate. The Theme Workspace is a starter to help you get started building a Gatsby Theme. 

This could possibly say something like:
- preconfigured theme and site to get started developing a theme
- yarn workspaces set up to connect the example site to the theme

File: starters/README.md [hunk 6]
```diff
@@ -60,11 +60,13 @@ A starter is a simplified example to get up and running with Gatsby quickly and
...
+|                          [hello-world](https://github.com/gatsbyjs/gatsby-starter-hello-world)                           | [gatsby-starter-hello-world-demo.netlify.com](https://gatsby-starter-hello-world-demo.netlify.com/)                           | A simplified bare-bones starter for Gatsby                                  |
+|      [gatsby-starter-blog-theme](https://github.com/gatsbyjs/gatsby/tree/master/starters/gatsby-starter-blog-theme)      | [gatsby-starter-blog-theme-docs](https://github.com/gatsbyjs/gatsby/tree/master/starters/gatsby-starter-blog-theme)           | A starter for a blog with augmented functionality provided by Gatsby themes |
+| [gatsby-starter-theme-workspace](https://github.com/gatsbyjs/gatsby/tree/master/starters/gatsby-starter-theme-workspace) | [gatsby-starter-theme-workspace-docs](https://github.com/gatsbyjs/gatsby/tree/master/starters/gatsby-starter-theme-workspace) | A sample starter utilizing a minimalist theme provided by Gatsby            |
```
Comment (gillkyle) [inline starters/README.md:69]: The description for the theme workspace should mention that it's to help you get started developing a theme.

```suggestion
| [gatsby-starter-theme-workspace](https://github.com/gatsbyjs/gatsby/tree/master/starters/gatsby-starter-theme-workspace) | [gatsby-starter-theme-workspace-docs](https://github.com/gatsbyjs/gatsby/tree/master/starters/gatsby-starter-theme-workspace) | A sample starter utilizing a minimalist theme provided by Gatsby to help you get started developing a theme           |
```

**Comment by marcysutton** [comment]
Hi @Edmund-Adewu, I wanted to follow up on this PR. Are you able to work in Kyle's feedback and resolve some merge conflicts?

**Comment by LekoArts** [comment]
As this PR has gone stale I'll close this. Please re-open or send in a new PR when you addressed the feedback, thanks!
